### PR TITLE
linking front thumbnails, updating screenshots + pyplot API page

### PR DESCRIPTION
**Pull request by choldgraf**
This links the frontpage thumbnails to the sphinx gallery versions so that we don't have to create them manually. Also moves `screenshots.rst` to a tutorial so continue the consolidation.

### CODE REVIEW

**Comment by story645** [comment]
:/ categorical doesn't work on numpy < 1.8 and I'll go fix that bug.

**Comment by QuLogic** [comment]
Do you have a screenshot of what it looks like? The main reason for the static images is that they don't look very good when scaled down (not that the existing ones are amazing examples in their own right.)

**Comment by choldgraf** [comment]
@QuLogic see the live version here:

http://predictablynoisy.com/matplotlib/

the only difference I can tell is that the images are slightly smaller, but the resolution is actually higher I think. 

@story645 it seems like the error is not related to this PR, no?

**Comment by story645** [comment]
Yeah, realized it wasn't and edited my comment as such...though did open #8592 to fix that bug since it was so long overdue.

**Comment by choldgraf** [comment]
woot!

File: tutorials/01_introductory/sample_plots.py [hunk 1]
```diff
@@ -373,30 +374,24 @@
...
    :scale: 50
 
    Xkcd
+"""
 
+# An extra set of sample plots for the thumbnail image.
+import matplotlib.pyplot as plt
+import numpy as np
 
+data = np.random.randn(2, 100)
```
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: please include a seed.

Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: please add seeds to random examples

File: examples/lines_bars_and_markers/categorical_variables.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Many times you want to create a plot that uses categorical variables
+in Matplotlib. For example, your data may naturally fall into
+several "bins" and you're interested in summarizing the data per
+bin. Matplotlib allows you to pass categorical variables directly to
+many plotting functions.
+"""
+import matplotlib.pyplot as plt
+
+names = ['group_a', 'group_b', 'group_c']
```
Comment (story645) [inline examples/lines_bars_and_markers/categorical_variables.py]: I'm in favor of more concrete examples. Here's some I did...and yes should have just added to matplotlib:
```python
data = {'apples':10, 'oranges':15, 'lemons':5, 'limes':20}
```
Also:
```python
cat = ["bored", "happy", "bored", "bored", "happy", "bored"]
dog = ["happy", "happy", "happy", "happy", "bored", "happy"]
time = ["combing", "drinking","feeding", "napping", "playing", "washing"]

fig = plt.figure()
ax = fig.add_subplot(1,1,1)
ax.plot(time, dog, label="dog")
ax.plot(time, cat, label="cat")
ax.legend()
```

Comment (choldgraf) [inline examples/lines_bars_and_markers/categorical_variables.py]: lol that's awesome, I'll replace w/ your example here, and maybe leave the tutorial as-is? Or add another example? WDYT?

**Comment by choldgraf** [comment]
ok - comments addressed

**Comment by choldgraf** [comment]
and tests passing!

File: examples/lines_bars_and_markers/categorical_variables.py (hunk 2 above)
Comment (story645) [inline examples/lines_bars_and_markers/categorical_variables.py]: I think two is good enough since categorical is supported by whatever methods are supported by pyplot and I don't even know that list.

File: tutorials/01_introductory/pyplot.py [hunk 3]
```diff
@@ -119,6 +119,26 @@
...
+# ===================================
+#
+# It is also possible to create a plot using categorical variables
+# Matplotlib allows you to pass categorical variables directly to
+# many plotting functions. For example:
+
+names = ['group_a', 'group_b', 'group_c']
+values = [1, 10, 100]
+
+fig, axs = plt.subplots(1, 3, figsize=(9, 3))
```
Comment (story645) [inline tutorials/01_introductory/pyplot.py]: this isn't the pyplot interface....

File: examples/lines_bars_and_markers/categorical_variables.py (hunk 2 above)
Comment (choldgraf) [inline examples/lines_bars_and_markers/categorical_variables.py]: cool - in the latest push I just have two little examples in the `example` script, and didn't change the tutorial

File: tutorials/01_introductory/pyplot.py (hunk 3 above)
Comment (choldgraf) [inline tutorials/01_introductory/pyplot.py]: ah yes - you're right...I never use `pyplot` so do it without thinking :)

**Comment by choldgraf** [comment]
latest push uses pyplot interface

File: tutorials/01_introductory/sample_plots.py [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
 
 Here you'll find a host of example plots with the code that
 generated them.
 
+.. _matplotlibscreenshots:
+
 Simple Plot
 ===========
 
 Here's a very basic :func:`~matplotlib.pyplot.plot` with text labels:
```
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: I like to avoid the word basic/simple/etc. in tutorials. What about "This is how to create a line plot with text labels"

File: tutorials/01_introductory/sample_plots.py [hunk 5]
```diff
@@ -58,8 +59,8 @@
 You can add arbitrary paths in matplotlib using the
```
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: What are arbitrary paths?

File: tutorials/01_introductory/sample_plots.py [hunk 6]
```diff
@@ -133,17 +134,17 @@
 Bar charts are simple to create using the :func:`~matplotlib.pyplot.bar`
```
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: Same about removing simple. `Use :func:`~matplotlib.pyplot.bar` command to make bar charts, which including customization:` (though won't hold you/hold up the pr to this since you didn't write it in the first place)

File: tutorials/01_introductory/sample_plots.py [hunk 7]
```diff
@@ -133,17 +134,17 @@
...
-.. figure:: ../gallery/pylab_examples/images/sphx_glr_barchart_demo_001.png
-   :target: ../gallery/pylab_examples/barchart_demo.html
+.. figure:: ../../gallery/pylab_examples/images/sphx_glr_barchart_demo_001.png
+   :target: ../../gallery/pylab_examples/barchart_demo.html
    :align: center
    :scale: 50
 
    Barchart Demo
 
 It's also simple to create stacked bars
```
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: same about simple

File: tutorials/01_introductory/sample_plots.py (hunk 5 above)
Comment (choldgraf) [inline tutorials/01_introductory/sample_plots.py]: not sure - I didn't create any of this text I just moved it here :-P  I'll call it "other paths" I guess

**Comment by choldgraf** [comment]
latest push removes the ableist language. The City of Berkeley would be proud of this PR now ;-)

File: tutorials/01_introductory/sample_plots.py (hunk 5 above)
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: nah, just leave it as is then and we can make it someone else's problem. I think the arbitrary is important.

**Comment by choldgraf** [comment]
I think the non-passing-travis build is not related to this PR

**Comment by choldgraf** [comment]
(dunno what's up with the appveyor build...)

**Comment by choldgraf** [comment]
can somebody restart those appveyor builds?

**Comment by QuLogic** [comment]
Done.

File: doc/api/pyplot_summary.rst [hunk 8]
```diff
@@ -1,8 +1,41 @@
...
 .. currentmodule:: matplotlib.pyplot
 
 .. autofunction:: plotting
 
+The Object-Oriented API
+-----------------------
+
+Most of these functions also exist as methods in the
+:class:`matplotlib.axes.Axes` class. You can use them with the
+so-called "Object Oriented" approach to Matplotlib.
```
Comment (story645) [inline doc/api/pyplot_summary.rst]: don't need so-called (it's redundant)

File: examples/lines_bars_and_markers/categorical_variables.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+==============================
+Plotting categorical variables
+==============================
+
+How to use categorical variables in matplotlib.
+
+Many times you want to create a plot that uses categorical variables
+in Matplotlib. For example, your data may naturally fall into
+several "bins" and you're interested in summarizing the data per
+bin. Matplotlib allows you to pass categorical variables directly to
```
Comment (story645) [inline examples/lines_bars_and_markers/categorical_variables.py]: Don't think you need this sentence since this is a histogram:
>For example, your data may naturally fall into several "bins" and you're interested in summarizing the data per bin.

File: tutorials/01_introductory/sample_plots.py [hunk 10]
```diff
@@ -243,8 +245,8 @@
 You can plot date data with major and minor ticks and custom tick formatters
```
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: You can plot timeseries data ('cause date data is hard to parse)

**Comment by choldgraf** [comment]
ok - I'll hold off to see if anybody else has suggestions, and if not then I'll make @story645 's edits and then merge?

File: tutorials/01_introductory/pyplot.py [hunk 11]
```diff
@@ -119,6 +119,30 @@
 plt.show()
 
 ###############################################################################
+# .. _plotting-with-categorical-vars:
+#
+# Plotting with categorical variables
+# ===================================
+#
+# It is also possible to create a plot using categorical variables
```
Comment (QuLogic) [inline tutorials/01_introductory/pyplot.py]: Missing period.

File: tutorials/01_introductory/sample_plots.py [hunk 12]
```diff
@@ -1,19 +1,21 @@
...
 
-Simple Plot
-===========
+.. _matplotlibscreenshots:
+
+Line Plot
+=========
 
-Here's a very basic :func:`~matplotlib.pyplot.plot` with text labels:
+Here's now to create a line plot with  text labels using
```
Comment (QuLogic) [inline tutorials/01_introductory/sample_plots.py]: Here's how?

File: tutorials/01_introductory/sample_plots.py [hunk 13]
```diff
@@ -55,11 +57,11 @@
 Path demo
 =========
 
-You can add arbitrary paths in matplotlib using the
+You can add other paths in matplotlib using the
```
Comment (QuLogic) [inline tutorials/01_introductory/sample_plots.py]: Matplotlib

File: examples/lines_bars_and_markers/categorical_variables.py [hunk 14]
```diff
@@ -0,0 +1,39 @@
+"""
+==============================
+Plotting categorical variables
+==============================
+
+How to use categorical variables in matplotlib.
```
Comment (QuLogic) [inline examples/lines_bars_and_markers/categorical_variables.py]: Matplotlib

File: doc/api/pyplot_summary.rst [hunk 15]
```diff
@@ -1,8 +1,41 @@
...
+of the :mod:`matplotlib.pyplot` module, see the
+:ref:`sphx_glr_tutorials_01_introductory_pyplot.py`
+or the :ref:`pyplots_examples`. We also recommend that you look into
+the object-oriented approach to plotting, described below.
 
 .. currentmodule:: matplotlib.pyplot
 
 .. autofunction:: plotting
 
+The Object-Oriented API
```
Comment (QuLogic) [inline doc/api/pyplot_summary.rst]: Should we put this above the Pyplot API section? It could be formatted as an aside/callout/warning if needed. But I think `autofunction` generates a big table of functions (your site appears to be rendering a different PR), so no-one will ever see this recommendation.

File: tutorials/01_introductory/sample_plots.py (hunk 13 above)
Comment (story645) [inline tutorials/01_introductory/sample_plots.py]: Also needs to be arbitrary cause the arbitrary is in reference to the shape of the part (line, circle, curve)

File: doc/api/pyplot_summary.rst (hunk 15 above)
Comment (choldgraf) [inline doc/api/pyplot_summary.rst]: ack yeah I'm sorry - I've got a buncha PRs open so it's hard to remember which one I've pushed to the website. I imagined just adding the section on OO stuff to the pyplot summary. It felt weird to put it at the beginning because the page is called `pyplot`. What if we instead called the page `plotting API` instead of `pyplot`? Then I could add the OO stuff first.

Comment (choldgraf) [inline doc/api/pyplot_summary.rst]: I should note - this is why I put the `.. contents::` bit at the top of the page - it'll render a table of contents that will list all the sections. I hoped that would make it easier to find the OO section (and the colormaps section...though TBH I'm still not really sure why that section is on this page ¯\_(ツ)_/¯

Comment (choldgraf) [inline doc/api/pyplot_summary.rst]: Just pushed this PR to the live version: http://predictablynoisy.com/matplotlib/api/pyplot_summary.html

**Comment by choldgraf** [comment]
ok comments addressed!

**Comment by choldgraf** [comment]
Looks like all tests are passing - the latest commit was just capitalizing one instance of `matplotlib` that I had missed...

**Comment by choldgraf** [comment]
anything to do here? ready to 🚢  or still need to ⚓️ ? :D

**Comment by story645** [comment]
ping @QuLogic

File: doc/api/pyplot_summary.rst (hunk 15 above)
Comment (QuLogic) [inline doc/api/pyplot_summary.rst]: We should probably think about this a bit in a later PR.

**Comment by choldgraf** [comment]
woo! :)
